agent/ec2/ec2_client.go

```diff
@@ -105,3 +108,21 @@ func (c *ClientImpl) DescribeECSTagsForInstance(instanceID string) ([]*ecs.Tag,
 func (c *ClientImpl) CreateTags(input *ec2sdk.CreateTagsInput) (*ec2sdk.CreateTagsOutput, error) {
 	return c.client.CreateTags(input)
 }
+
+// GetHostPublicIPv4AddressForInstance calls the DescribeInstances API to get the
+// PublicIPv4 address for a specific instance. Only 1 address is expected.
+func (c *ClientImpl) GetHostPublicIPv4AddressForInstance(instanceID string) (string, error) {
+	describeInstancesInput := ec2sdk.DescribeInstancesInput{
+		InstanceIds: []*string{aws.String(instanceID)},
+	}
+	describeInstancesOutput, err := c.client.DescribeInstances(&describeInstancesInput)
+	if err != nil {
+		seelog.Criticalf("Error calling DescribeInstances API: %v", err)
+		return "", err
+	}
+	if len(describeInstancesOutput.Reservations) == 0 || len(describeInstancesOutput.Reservations[0].Instances) == 0 {
```

Comment:
`Only 1 address is expected.`, if only 1 is expected. should we explicitly check for 1 for each? am i missing something?

Reply:
Because we only ever pass in a single instance ID, I can't imagine a scenario in which we get more than 1 instance back. This check is to ensure we don't run into a nil pointer exception